### ISSUE TITLE
Contributor info change requests

### ISSUE THREAD

**Issue by jfmengels** [kentcdodds/all-contributors#19]
I'm wondering whether the spec should describe the behavior of the maintainers when a contributor asks to change his info (let's say by submitting a PR with the updates).

For example, if I was a contributor of a given project two years ago, and want to
- change my avatar
- the url my avatar links to
- add the link to a video of a talk I made on the
- remove a link
- change my name
- remove myself from the list

Should the maintainer be "obligated" to accept such a change? Do we want the spec to have a position on this? What if there is some obscene wording/images/links in what is demanded?
Thoughts?

**Comment by kentcdodds** [comment]
I agree that it'd be really silly if a maintainer rejected a request to change info. I don't think we could really enforce anything anyway. If someone were being a jerk enough to not accept a change, they'd probably not follow our instructions to accept the change anyway...

Thoughts?

**Comment by dhairyabaweja** [comment]
I think it would not be a good idea for the maintainer to control such info changes.

**Comment by Berkmann18** [comment]
I don't think GDPR is of much influence on repos (other than on GitHub itself) but when it comes to data access/modification/deletion requests, I think maintainers **should** comply with that.

Basically, one of the things GDPR states is that if a data subject (in this case a contributor) perform a data access/modification/... request, the company/service that holds data on the data subject (in this case: avatar, URL on the profile, name, ...) has to comply with that.

In any case, as @kentcdodds said, it would be silly if a maintainer rejected that (especially if the request is legitimate).

**Comment by MatheusRV** [comment]
How about ask for help in GitHub Team?

**Comment by Berkmann18** [comment]
@MatheusRV Good point, I've contacted them to see what they think.

**Comment by Berkmann18** [comment]
> It is the repository controller's responsibility to determine how to handle data subject requests under the GDPR or other data privacy laws, and GitHub cannot provide legal advice or assistance on users' data privacy compliance.
> -- <cite>Jules Parker (Github Staff)</cite>

**Comment by sirpeas** [comment]
In my opinion, this issue is very area of improvement according to GDPR, especially when "controller of repository" is responsible for GDPR related requests to be done. I think that new commands to the bot/cli could partly solve that issue.

A person who wants to do some modification or even remove himself from contributors should be an initiator of those actions by new PR or contacting the owner and then maintainer is responsible for making that happen 😄.

cc @jakebolam @sinchang @Berkmann18

**Comment by Berkmann18** [comment]
@sirpeas Having a cli/bot command for that seems nice; however, I think it might be overkill (**except for the data removal request**) considering how unlikely and rare situations like those are so for the most part. the data subject should either submit a PR with the relevant changes or open an issue or contact one of the maintainers.

**Comment by simlrh** [comment]
As someone in this situation, I don't mind having to manually submit a PR removing myself as a contributor, but some documentation about what needs to change would be appreciated.

All I knew was that google turned up some personal information in someone else's README and I wanted it gone. It's taken lots of googling and digging through docs to find out what `all-contributors` is and how it works, and what I would need to delete in the PR to make that happen, and I still don't know if it will work.

I believe I need to remove my entry from `.all-contributorsrc`, but I don't think doing so will actually trigger the bot to regenerate the README, so I also need to delete my section from `README.md`, even though there's a comment saying not to modify it. Is that correct?

**Comment by Berkmann18** [comment]
> As someone in this situation, I don't mind having to manually submit a PR removing myself as a contributor, but some documentation about what needs to change would be appreciated.

That's a great point, and I'm sure having a section on **DRRs (Data Removal Requests)/DCRs (Data Change Requests)** would be beneficial. If you've got something in the works, feel free to submit a (draft) PR or get hold of me on the Slack channel and I could help (I may not be as responsive as I used to be tho due to some life-changing events tho).

> I believe I need to remove my entry from .all-contributorsrc, but I don't think doing so will actually trigger the bot to regenerate the README

AFAIK, the bot isn't equipped to deal with DRRs so such operations need to be done through the CLI or manually followed by a `all-contributors generate` execution on the relevant repo.

> I believe I need to remove my entry from `.all-contributorsrc`, but I don't think doing so will actually trigger the bot to regenerate the README, so I also need to delete my section from `README.md`, even though there's a comment saying not to modify it. Is that correct?

If you _only_ modify the `README.md` then the second `all-contributors generate` is ran on the same repo (assuming its configured to modify `README.md`) or if the bot does it then your changes will be gone (unless you've also modified the `.all-contributorsrc`).
